cmd/state-svc/test/integration/svc_int_test.go

```diff
@@ -31,20 +34,37 @@ func (suite *SvcIntegrationTestSuite) TestStartStop() {
 	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("start"))
 	cp.Expect("Starting")
 	cp.ExpectExitCode(0)
+	time.Sleep(500 * time.Millisecond) // wait for service to start up
 
 	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
 	cp.Expect("Checking")
-	cp.ExpectRe("Port:\\s+:\\d+")
-	cp.ExpectRe("Log:\\s+.+?\\.log")
-	cp.ExpectExitCode(0)
 
+	// Verify it created a socket file.
 	sockFile := svcctl.NewIPCSockPathFromGlobals().String()
 	suite.True(fileutils.TargetExists(sockFile))
 
+	// Verify the server is running on its reported port.
+	cp.ExpectRe("Port:\\s+:\\d+")
+	portRe := regexp.MustCompile("Port:\\s+:(\\d+)")
+	port := portRe.FindStringSubmatch(cp.TrimmedSnapshot())[1]
+	_, err := net.Listen("tcp", "localhost:"+port)
+	suite.Error(err)
+
+	// Verify it created and wrote to its reported log file.
+	cp.ExpectRe("Log:\\s+.+?\\.log")
+	logRe := regexp.MustCompile("Log:\\s+(.+?\\.log)")
+	logFile := logRe.FindStringSubmatch(cp.TrimmedSnapshot())[1]
+	suite.True(fileutils.FileExists(logFile))
+	suite.True(len(fileutils.ReadFileUnsafe(logFile)) > 0)
+
+	cp.ExpectExitCode(0)
+
 	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("stop"))
 	cp.Expect("Stopping")
 	cp.ExpectExitCode(0)
+	time.Sleep(500 * time.Millisecond) // wait for service to stop
 
+	// Verify it deleted its socket file.
```

Comment:
Let's also verify here that the port is now available. That should help surface any sort of edge cases where listening on the port was failing due to other reasons (eg. firewall).